### PR TITLE
Enable message tagging for outbound messaging events

### PR DESCRIPTION
**Pull request by miguelpdiaz8**
[CHANNELS-801](https://segment.atlassian.net/browse/CHANNELS-801) This PR:

- Remove statusCallbackUrl for outbound messages
- Add tags for required attribute information
- Add unit tests

## Testing

In progress

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

[CHANNELS-801]: https://segment.atlassian.net/browse/CHANNELS-801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
hi @miguelpdiaz8 thanks for raising this PR. I'll schedule for review.

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts [hunk 1]
```diff
@@ -40,7 +42,25 @@ export abstract class PhoneMessageSender<Payload extends PhoneMessagePayload> ex
       this.DEFAULT_CONNECTION_OVERRIDES
     )
 
-    if (webhookUrlWithParams) body.append('StatusCallback', webhookUrlWithParams)
+    if (
+      this.executeInput &&
+      this.executeInput.features &&
+      this.executeInput.features[FLAGON_EVENT_STREAMS_ONBOARDING]
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:48]: `executeInput` will always be defined.
`this.executeInput.features?.[FLAGON_EVENT_STREAMS_ONBOARDING]`? <-- syntax sugar

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts [hunk 2]
```diff
@@ -40,7 +42,25 @@ export abstract class PhoneMessageSender<Payload extends PhoneMessagePayload> ex
...
+    ) {
+      const tags = {
+        audience_id: this.payload.customArgs ? String(this.payload.customArgs['audience_id']) : '',
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:51]: Is it necessary to cast these to strings? Not sure if its expected by Twilio or not, but I'd suspect that we'd want the data type passed by personas service to be the data type returned in the event stream response.

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts [hunk 3]
```diff
@@ -40,7 +42,25 @@ export abstract class PhoneMessageSender<Payload extends PhoneMessagePayload> ex
...
+        journey_name: this.payload.customArgs ? String(this.payload.customArgs['journey_name']) : '',
+        step_name: this.payload.customArgs ? String(this.payload.customArgs['step_name']) : '',
+        campaign_name: this.payload.customArgs ? String(this.payload.customArgs['campaign_name']) : '',
+        campaign_key: this.payload.customArgs ? String(this.payload.customArgs['campaign_key']) : '',
+        user_id: this.payload.customArgs ? String(this.payload.customArgs['user_id']) : '',
+        message_id: this.payload.customArgs ? String(this.payload.customArgs['message_id']) : ''
+      }
+      body.append('tags', JSON.stringify(tags))
+    } else {
+      if (webhookUrlWithParams) body.append('StatusCallback', webhookUrlWithParams)
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:62]: would `else if (webhookUrlWithParams) { body.append('StatusCallback', webhookUrlWithParams) }` work here?

**Review by alfrimpong — COMMENTED** [review]
Looks good, just minor comments

**Comment by joe-ayoub-segment** [comment]
hi @alfrimpong @miguelpdiaz8 can you notify me once you are ready for me to review the PR please? I see some comments from @alfrimpong which still need to be responded to.

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts (hunk 1 above)
Comment (miguelpdiaz8) [reply packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts]: Done, thanks!

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts (hunk 2 above)
Comment (miguelpdiaz8) [reply packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts]: Updated, thanks!

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts [hunk 4]
```diff
@@ -40,7 +42,25 @@ export abstract class PhoneMessageSender<Payload extends PhoneMessagePayload> ex
...
+        audience_id: this.payload.customArgs ? String(this.payload.customArgs['audience_id']) : '',
+        correlation_id: this.payload.customArgs ? String(this.payload.customArgs['correlation_id']) : '',
+        journey_name: this.payload.customArgs ? String(this.payload.customArgs['journey_name']) : '',
+        step_name: this.payload.customArgs ? String(this.payload.customArgs['step_name']) : '',
+        campaign_name: this.payload.customArgs ? String(this.payload.customArgs['campaign_name']) : '',
+        campaign_key: this.payload.customArgs ? String(this.payload.customArgs['campaign_key']) : '',
+        user_id: this.payload.customArgs ? String(this.payload.customArgs['user_id']) : '',
+        message_id: this.payload.customArgs ? String(this.payload.customArgs['message_id']) : ''
+      }
+      body.append('tags', JSON.stringify(tags))
```
[the remark this answers is not in the document]
Comment (miguelpdiaz8) [reply packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:56]: tags seems to be correct, thanks!

**Comment by joe-ayoub-segment** [comment]
hi @alfrimpong and @miguelpdiaz8 - just a reminder to ping me once you want me to review + deploy.

**Comment by alfrimpong** [comment]
> hi @alfrimpong and @miguelpdiaz8 - just a reminder to ping me once you want me to review + deploy. If you want me to ignore the PR for now please make it a draft.

Will do, we're doing the final end to end testing for this right now

**Comment by alfrimpong** [comment]
@joe-ayoub-segment This is ready for review

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @miguelpdiaz8 and @alfrimpong  this PR has been deployed.

**Pull request merged**